What's New 104 & updated Experiments (Stable 103, Canary 105) (#2072)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2022/07/devtools-104.md

<!-- Subtitle: Layers isn't a stand-alone tool anymore, it's wrapped up in the comprehensive 3D View; a tool for all your visual debugging needs. -->

The standalone **Layers** tool has been removed in Microsoft Edge version 104.  In previous versions of Microsoft Edge, the **Layers** tool redirected users to the **3D View** tool.

In the **3D View** tool, the **Composited Layers** view has all the features you love from the **Layers** tool, along with some updates.  The **Composited Layers** view is now complemented by two other visual debugging panes: **DOM** and **Z-index**.  Combine these additional views with the **Composited Layers** view for better debugging.

![The Composited Layers tab in the 3D View tool](./devtools-104-images/3d-view-composited-layers.png)

See also:
* [Debug the web in 3D with the 3D View tool](https://blogs.windows.com/msedgedev/2022/06/21/debug-the-web-in-3d-with-the-3d-view-tool/)
* [Navigate webpage layers, z-index, and DOM using the 3D View tool](../../../3d-view/index.md)

<!-- ====================================================================== -->
## Request URLs in Issues tool have been shortened to display better

> 

Reviewer: Title should be "In Issues tool, long request URLs have been shortened" as shown in the metadata below.

Author: Made heading match "Title:" comment tho usu want to be different

![Network tool](./devtools-104-images/network-tool-ws-arrows.png)
<!-- Start by using the work item attached png

1. Navigate to: Online WebSocket & Socket.io Tester - PieSocket Realtime
2. Open DevTools > Network and undock the tools
3. On the website, select Connect and send a test message
4. In the Network tool, select WS to filter to only WebSocket network requests
5. Select the one network request that should appear
6. Take a screenshot and draw a red highlight box over the red and green arrows -->

See also:
* [Network features reference](../../../network/reference.md)

<!-- ====================================================================== -->
## Announcements from the Chromium project

Microsoft Edge 104 also includes the following updates from the Chromium project:

* [Restart frame during debugging](https://developer.chrome.com/blog/new-in-devtools-104/#restart-frame)
* [Slow replay options in the Recorder panel](https://developer.chrome.com/blog/new-in-devtools-104/#recorder)

> * [Build an extension for the Recorder panel](https://developer.chrome.com/blog/new-in-devtools-104/#recorder-extension)

Reviewer 2: The recorder panel exist only in Chrome. It is not part of the Chromium open source project, and hence, not available in Edge. So I don't think we should include it in this Chromium project section.

Reviewer 3: Has this been resolved yet @mikehoffms?

Author: @erdraud - yes, removed those two links.  closing thread as Resolved.  
Rendered: https://review.docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2022/07/devtools-104?branch=pr-en-us-2072#announcements-from-the-chromium-project

---
title: What's new in DevTools (Microsoft Edge 104)
description: "Use Layers in 3D View. Improved display of request URLs in Issues tool. Improving accessibility for the Detached Elements tool. Fix: Arrows representing web socket traffic now render correctly in the Network tool. And more."
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 07/21/2022
---
# What's New in DevTools (Microsoft Edge 104)

[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]

<!-- ====================================================================== -->
## Use Layers in 3D View

<!-- todo: check work item -->

<!-- Title: Where did Layers go? -->
<!-- Subtitle: Layers isn't a stand-alone tool anymore, it's wrapped up in the comprehensive 3D View; a tool for all your visual debugging needs. -->

> The standalone **Layers** tool has now been completely removed.  During the phase-out period, the **Layers** tool redirected users to the **3D View** tool.

Reviewer 4: I don't think we need to go into so much detail about the removal process itself. See my suggestion:

```suggestion
The standalone **Layers** tool has been removed in MIcrosoft Edge version 104.  In previous versions of Microsoft Edge, the **Layers** tool redirected users to the **3D View** tool.
```

Author: fixed cap'zn

> In the **3D View** tool, the **Composited Layers** view has all the features you love from the **Layers** tool, along with some updates.  The **Composited Layers** view is now complemented by two other visual debugging panes.  Try out the **DOM** and **Z-index** views for debugging while you're in the **3D View** tool, and you'll see how these three views work together.

Reviewer 4: ```suggestion
In the **3D View** tool, the **Composited Layers** view has all the features you love from the **Layers** tool, along with some updates.  The **Composited Layers** view is now complemented by two other visual debugging panes: **DOM** and **Z-index**.  Combine these additional views with the **Composited Layers** view for better debugging.
```

![The Composited Layers tab in the 3D View tool](./devtools-104-images/3d-view-composited-layers.png)

See also:

> * [Navigate webpage layers, z-index, and DOM using the 3D View tool](../../../3d-view/index.md)

Reviewer 2: It would also be good to link to our recent blog post about this: https://blogs.windows.com/msedgedev/2022/06/21/debug-the-web-in-3d-with-the-3d-view-tool/

Reviewer 4: +1, let's add this link as well

Author: * [Debug the web in 3D with the 3D View tool](https://blogs.windows.com/msedgedev/2022/06/21/debug-the-web-in-3d-with-the-3d-view-tool/)

<!-- ====================================================================== -->

> ## Improved display of request URLs in Issues tool

Reviewer 4: Feel like we have too many "Improved" vs. "Improving" titles this release. Here's a suggestion for this section but we could also leave this one as-is and change the next one. Up to you!

```suggestion
## Request URLs in Issues tool have been shortened to display better
```

<!-- Title: In Issues tool, long request URLs have been shortened -->
<!-- Subtitle: Links to network request URLs have been truncated, making issues easier to read. -->

> The **Issues** tool provides recommendations to improve a web page's accessibility, performance, security, and more.  Some of the recommendations that are listed include a link to a network request that opens in the **Network** tool.

Reviewer 4: ```suggestion
The **Issues** tool provides recommendations to improve a web page's accessibility, performance, security, and more.  Some of these recommendations around networking include a link to a given request that opens in the **Network** tool.
```

> Previously, issues related to network requests displayed the entire URL for the network request.  When these URLs were lengthy, they took up a lot of space, making the list of issues difficult to browse.  In Microsoft Edge 104, these long request URL links have been truncated.  The links still direct to the relevant part of DevTools, but now take up less space.

Reviewer 4: ```suggestion
Previously, issues related to network requests displayed the entire URL for the network request.  When these URLs were lengthy, they took up a lot of space, making the list of issues difficult to browse.  In Microsoft Edge 104, these long request URL links have been truncated to take up less space.  The links still direct to the relevant part of the **Network* tool.
```

Author: added the 2nd *

> Before:

Reviewer 4: ```suggestion
In previous versions of Microsoft Edge:
```

![Issues tool showing a long URL, making the issue entry too tall](./devtools-104-images/issues-request-urls-long.png)

> After:

Reviewer 4: ```suggestion
In Microsoft Edge version 104:
```

---
title: What's new in DevTools (Microsoft Edge 104)
description: "Use Layers in 3D View. Improved display of request URLs in Issues tool. Improving accessibility for the Detached Elements tool. Fix: Arrows representing web socket traffic now render correctly in the Network tool. And more."
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 07/21/2022
---
# What's New in DevTools (Microsoft Edge 104)

[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]

<!-- ====================================================================== -->
## Use Layers in 3D View
<!-- rank: 1 -->

<!-- todo: check work item -->

<!-- Title: Where did Layers go? -->
<!-- Subtitle: Layers isn't a stand-alone tool anymore, it's wrapped up in the comprehensive 3D View; a tool for all your visual debugging needs. -->

> The standalone **Layers** tool has now been completely removed.  During the phase-out period, the **Layers** tool remained, as an empty panel containing a link that opened the **3D View** tool.

Reviewer 3: During the phase-out period, the **Layers** tool redirected users to the **3D View** tool.

> In the **3D View** tool, the **Composited Layers** view has all the features you love from the **Layers** tool, along with some updates.  The **Composited Layers** view now sits alongside even more visual debugging tools.  Try out the **DOM** and **Z-index** views for debugging while you're in the **3D View** tool, and you'll see how these three views work together.

Reviewer 3: The **Composited Layers** view is complemented by two other visual debugging panes.
